cogs/birthday/converters.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime
+from redbot.core import commands
+
+# Using February 29 or February 3 below, but %b, %B will be in the bot's
+# locale.
+# %Y in all cases will be a leap year (2020), omitting from examples
+FORMAT_TUPLE: tuple = (
+    "%m %d %Y",  # 2 29, 02 29, 02 03, 2 3, etc.
+    "%b %d %Y",  # Feb 29, Feb 03, Feb 3
+    "%d %b %Y",  # 29 Feb, 03 Feb, 3 Feb
+    "%B %d %Y",  # February 29, February 3, February 03
+    "%d %B %Y",  # 29 February, 3 February, 03 February
+)
+
+
+class MonthDayConverter(commands.Converter):
+    async def convert(self, ctx: commands.Context, dateString: str) -> datetime:
+        inputDate = f"{dateString} 2020"
+        for formatString in FORMAT_TUPLE:
+            try:
+                dateObj: datetime = datetime.strptime(inputDate, formatString)
+            except ValueError:
+                print(f"Error {formatString}, {inputDate}")
+            else:
+                return dateObj
+        raise commands.BadArgument()
```

Comment:
Thoughts on using `dateutil` (pip `python-dateutil`) for the parsing work? It seems like we are doing some manual work here by defining a set of date formats. A piece of tl;dr about `dateutil` is that passing `"Dec 25"` to its parser would give us a `datetime` object for December 25, 2022, and `"Feb 29"` would raise a `ValueError` exception. With that we just need to simply handle.

Reply:
Is it possible to only parse dates and not times? i.e. `[p]birthday add <user> 12:30` would succeed in this case, but we should really be showing the help because that's not a date: it just so happens we default to the current date when specifying just the time.

Reply:
As per our side discussion, will check for non-zero hour, min, sec, millisec. Accepting 12am/00:00 is okay.